File: 30-Day LeetCoding Challenge/Python/backspaceCompare.py
class Solution(object):
    def getString(self, S):
        ans = ""
        count = 0
        for i in range(len(S)-1, -1, -1):
            if S[i] == "#":
                count += 1
            else:
                if count == 0:
                    ans = S[i] + ans
                else:
                    count -= 1
        return ans
    def backspaceCompare(self, S, T):
        """
        :type S: str
        :type T: str
        :rtype: bool
        """
        if self.getString(S) == self.getString(T):
            return True
        return False
s = Solution()
S = "ab##"
T = "c#d#"
print(s.backspaceCompare(S, T))

File: 30-Day LeetCoding Challenge/C++/backspaceCompare.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    string stringInEditor(string S) {
        string str_in_editor;
        int count = 0;
        for (int i = S.length() - 1; i >= 0; i--) {
			if (S[i] == '#') {
				count++;
			}
			else {
				if (count == 0) {
					str_in_editor = S[i] + str_in_editor;
				}
				else count--;
			}
		}
        return str_in_editor;
    }
	bool backspaceCompare(string S, string T) {
		return stringInEditor(S) == stringInEditor(T) ? true : false;

	}
};

int main() {
    Solution s;
    string S = "ab##";
    string T = "c#d#";
    cout << s.backspaceCompare(S, T);
    return 0;
}